test/integration/volume/persistent_volumes_test.go

```diff
@@ -1036,6 +1040,164 @@ func TestPersistentVolumeMultiPVsDiffAccessModes(t *testing.T) {
+			Name: defaultStorageClassName,
+			Annotations: map[string]string{
+				util.IsDefaultStorageClassAnnotation: "true",
+			},
+		},
+		Provisioner: provisionerPluginName,
+	}
+	if _, err := testClient.StorageV1().StorageClasses().Create(context.TODO(), &defaultSC, metav1.CreateOptions{}); err != nil {
+		t.Errorf("Failed to create a storage class: %v", err)
+	}
+
+	// 2. Test binding PV with storage class "" to PVC with storage class "" while default exists.
+	pv2 := createPVWithStorageClass("pv-2", "/tmp/bar", "5G", "",
+		[]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}, v1.PersistentVolumeReclaimRetain)
+	_, err = testClient.CoreV1().PersistentVolumes().Create(context.TODO(), pv2, metav1.CreateOptions{})
+	if err != nil {
+		t.Errorf("Failed to create PersistentVolume: %v", err)
+	}
+
+	pvc2 := createPVC("pvc-2", ns.Name, "5G", []v1.PersistentVolumeAccessMode{v1.ReadWriteMany}, "")
+	_, err = testClient.CoreV1().PersistentVolumeClaims(ns.Name).Create(context.TODO(), pvc2, metav1.CreateOptions{})
+	if err != nil {
+		t.Errorf("Failed to create PersistentVolumeClaim: %v", err)
+	}
+
+	// wait until the controller pairs the volume
+	waitForPersistentVolumePhase(testClient, pv2.Name, watchPV, v1.VolumeBound)
+	t.Log("volume bound")
+	waitForPersistentVolumeClaimPhase(testClient, pvc2.Name, ns.Name, watchPVC, v1.ClaimBound)
+	t.Log("claim bound")
+
+	pv, err = testClient.CoreV1().PersistentVolumes().Get(context.TODO(), "pv-2", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Unexpected error getting pv: %v", err)
+	}
+	if pv.Spec.ClaimRef == nil {
+		t.Fatalf("PV %s with \"\" storage class should have been bound to PVC %s that also has \"\" storage class", pv2.Name, pvc2.Name)
+	}
+	if pv.Spec.ClaimRef.Name != pvc2.Name {
+		t.Fatalf("Bind mismatch! Expected %s but got %s", pvc2.Name, pv.Spec.ClaimRef.Name)
+	}
+
+	// 3. Test PV with storage class "" does not bind to PVC with nil while default SC exists.
+	pv3 := createPVWithStorageClass("pv-3", "/tmp/bar", "5G", "",
+		[]v1.PersistentVolumeAccessMode{v1.ReadWriteMany}, v1.PersistentVolumeReclaimRetain)
+	_, err = testClient.CoreV1().PersistentVolumes().Create(context.TODO(), pv3, metav1.CreateOptions{})
+	if err != nil {
+		t.Errorf("Failed to create PersistentVolume: %v", err)
+	}
+
+	pvcRetro := createPVCWithNilClass("pvc-provision-noclass", ns.Name, "1G", []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce})
+	if _, err := testClient.CoreV1().PersistentVolumeClaims(ns.Name).Create(context.TODO(), pvcRetro, metav1.CreateOptions{}); err != nil {
+		t.Errorf("Failed to create PVC: %v", err)
+	}
+	t.Log("claim created")
+
+	// wait for pv to be created
+	waitForPersistentVolumePhase(testClient, pv3.Name, watchPV, v1.VolumePending)
+
+	pv, err = testClient.CoreV1().PersistentVolumes().Get(context.TODO(), "pv-3", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Unexpected error getting pv: %v", err)
+	}
+	if pv.Spec.ClaimRef != nil {
+		t.Fatalf("PV %s shouldn't be bound", pv3.Name)
+	}
```

Comment:
Move this check below after PVC is bound - at this time, a buggy PV controller might not have enough time to bind the bad PV.

Reply:
Done.